File: src/main/linkedList/LinkedList.java
package main.linkedList;

import main.linkedList.exceptions.EmptyListException;

public class LinkedList<TYPE> {

    private Node<TYPE> begin;

    private Integer size;


    public LinkedList() {
        this.begin = null;
        this.size = 0;
    }

    public void add(TYPE element){
        Node<TYPE> newElement = new Node<>();
        newElement.setElement(element);
        newElement.setNext(null);
        if(begin == null){
            begin = newElement;
        }else {
            Node<TYPE> aux;
            aux = begin;
            while(aux.getNext() != null){
                aux = aux.getNext();
            }
            aux.setNext(newElement);
        }
        size++;
    }

    public TYPE delete(){
        TYPE element = begin.getElement();
        if(!isEmpty()){
            Node<TYPE> aux = begin.getNext();
            this.begin = aux;
            this.size--;
            return element;
        }else {
            throw new EmptyListException("The list is empty, add something.");
        }
    }

    public void list(){
        if(begin == null){
            System.out.println("List is empty");
        }else {
            Node<TYPE> aux = begin;
            int i = 1;
            while (aux != null){
                System.out.println("Element "+ i + " "+ aux.getElement());
                aux = aux.getNext();
                i++;
            }
        }
    }
    public Boolean isEmpty(){
        return this.begin == null;
    }

    public Integer size(){
        return this.size;
    }

}

File: src/main/linkedList/c/linkedList.c
#include "linkedList.h"
#include <stdlib.h>

void create(struct List *list){
    list->begin = NULL;
}

void add(struct List *list, int element){
    struct Node *new;
    struct Node *aux;
    new = (struct Node*) malloc(sizeof(struct Node));
    new->next = NULL;
    new->element = element;
    if(list->begin == NULL){
        list->begin = new;
    }else{
        aux = list->begin;
        while(aux->next != NULL){
            aux = aux->next;
        }
        aux->next = new;

    }

}
int isEmpty(struct List list){
    return (list.begin == NULL);
}
int removee(struct List *list){
    struct Node *aux = list->begin;
    int element = aux->element;
    list->begin = aux->next;
    free(aux);
    aux = NULL;
    return element;
}
void listt(struct List list){
}